### PR TITLE
[14.0][NEW] sale_coupon_fixed_price: new reward type - fixed price

### PR DESCRIPTION
**Pull request by ivs-cetmix**
Sales Coupon Fixed Price

### CODE REVIEW

**Review by luis-ron — APPROVED** [review]
Functional review, LGTM

**Review by francesco-ooops — CHANGES_REQUESTED** [review]
@GabbasovDinar @isserver1 if reward is = fixed price, field "quantity" should not appear

image

Also, please add in USAGE file: "Please be careful when using this type of reward with "Discount apply on" options "On Order" or "On Cheapest Product" as auxiliary products (such as shipping fees, free products, etc) price can be easily overwritten by mistake by the promotion."

there is also a "Preice" typo in this line: https://github.com/OCA/sale-promotion/pull/55/files#diff-4143e5b656bb706594494bb0d065fe3fb190779a63b15806639093f84642a112R5

**Comment by GabbasovDinar** [comment]
@francesco-ooops fixed

**Review by francesco-ooops — APPROVED** [review]
@GabbasovDinar great, LGTM

**Comment by elvise** [comment]
@OCA/crm-sales-marketing-maintainers please check this PR :)

**Comment by pedrobaeza** [comment]
I don't have very clear the purpose of the module.

**Comment by ivs-cetmix** [comment]
> 

The idea is to add an option to use set fixed price for the product.
For example "If domain is matched then set the price 9.99 for the selected product"

**Comment by pedrobaeza** [comment]
Got it. I suggest as module name `sale_coupon_reward_fixed_price`

File: sale_coupon_fixed_price/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+This module adds an option to set a fixed price for condition matching products
```
Comment (pedrobaeza) [inline sale_coupon_fixed_price/readme/DESCRIPTION.rst]:
```suggestion
This module adds an option to set a fixed price for the matching line when the coupon is applied.
```

**Comment by francesco-ooops** [comment]
> For example "If domain is matched then set the price 9.99 for the selected product"

@pedrobaeza With standard behavior, a more fitting example is "if conditions for applying promotion are met, you can buy this extra product for only 1€ (instead of standard price)"

image

This feature can be also used by applying the fixed price on "cheapest product" to fulfill promotion type eg. "if you buy 3 products, pay the cheapest 1€".

In description we decided to warn user to be careful with "cheapest product" option, because it can easily be applied to shipping, advance payment, etc. We're planning a future module to specify with a domain which products should be excluded from such a promotion.

**Comment by pedrobaeza** [comment]
Coupons can be also applied through a code with no conditions, so you should develop the module for both cases.

**Comment by francesco-ooops** [comment]
> Coupons can be also applied through a code with no conditions, so you should develop the module for both cases.

can you expand on this? Are you referring to the future module or to this one?

**Comment by pedrobaeza** [comment]
It's a native feature. Go to "Coupon Programs" instead of " Promotion Programs".

**Comment by francesco-ooops** [comment]
> It's a native feature. Go to "Coupon Programs" instead of " Promotion Programs".

Yes, I'm aware of that :) I just wanted to make sure I understood your previous suggestion, can you clarify it?

**Comment by pedrobaeza** [comment]
I have just pointed a more neutral redaction of the module description + module name according to the stated.

**Comment by francesco-ooops** [comment]
> I have just pointed a more neutral redaction of the module description + module name according to the stated.

got it now, thanks!

**Comment by elvise** [comment]
@pedrobaeza Fixed! Can we merge? :)

**Comment by ivs-cetmix** [comment]
New PR with all fixes: https://github.com/OCA/sale-promotion/pull/59

**Pull request closed without merging**